File: sem2/po/PO-VirtualWorld-Python/simulator/organisms/animals/Wolf.py
from simulator.organisms.animals.Animal import Animal, BASE_ANIMAL_SPEED
from simulator.organisms.Organism import ORGANISM_TYPES

WOLF_STRENGTH = 9
WOLF_INITIATIVE = 5
WOLF_SPEED = BASE_ANIMAL_SPEED


class Wolf(Animal):
    def __init__(self, new_world, new_location):
        super().__init__(new_world, new_location, WOLF_STRENGTH, WOLF_INITIATIVE, ORGANISM_TYPES["Wolf"], WOLF_SPEED)

    def spawn_offspring(self, new_location):
        return Wolf(self.world, new_location)

    def __str__(self):
        return "Wolf at " + str(self.location)

File: sem2/po/po_proj2/src/studia/prog_ob/projekt2/world_simulator/organisms/animals/animals_subclasses/Wolf.java
package studia.prog_ob.projekt2.world_simulator.organisms.animals.animals_subclasses;


import studia.prog_ob.projekt2.world_simulator.World;
import studia.prog_ob.projekt2.world_simulator.maps.MapLocation;
import studia.prog_ob.projekt2.world_simulator.organisms.Organism;
import studia.prog_ob.projekt2.world_simulator.organisms.animals.Animal;

public class Wolf extends Animal {

    public static final int WOLF_STRENGTH = 9;
    public static final int WOLF_INITIATIVE = 5;
    public static final int WOLF_SPEED = BASE_ANIMAL_SPEED;
    public static final char WOLF_SYMBOL = 'W';

    public Wolf(MapLocation new_map_location, World new_world) {
        super(WOLF_STRENGTH, WOLF_INITIATIVE, new_map_location, WOLF_SYMBOL, new_world, WOLF_SPEED);
    }

    @Override
    public Organism spawn_offspring(MapLocation location) {
        return getWorld().add_organism(new Wolf(location, getWorld()));
    }

    @Override
    public String toString() {
        return "Wolf " + getMap_location() + " ";
    }
}
